### ISSUE TITLE
systemd-repart fails to run with "Not enough disk space" after initial A/B update

### ISSUE THREAD

**Issue by daandemeyer** [systemd/systemd#34528]
### systemd version the issue has been seen with

main

### Used distribution

Arch Linux

### Linux kernel version used

N/A

### CPU architectures issue was seen on

x86_64

### Expected behaviour you didn't see

- After setting up an A/B updated system with systemd-repart, On subsequent boots systemd-repart runs are a noop.

### Unexpected behaviour you saw

- systemd-repart fails to run with "Not enough disk space to fit requested partitions" on subsequent boots after the B partitions are populated by systemd-repart.

### Steps to reproduce the problem

- Set up A/B updated system with systemd-repart. Specifically, Add B partitions on first boot with systemd-repart all having the `_empty` label.
- Install an initial update that populates the B partitions with systemd-sysupdate
- Reboot the system
- systemd-repart will fail on reboot with "Not enough disk space to fit requested partitions"

The error happens because the B partition definitions contain `Label=_empty` which doesn't match any existing partitions anymore after the first update as the labels of the B partitions will have been changed by systemd-sysupdate so systemd-repart tries to create new B partitions for which there isn't any space.

If there was still free space left, systemd-repart would create new B partitions after every reboot after an update with systemd-sysupdate.

**Comment by poettering** [comment]
hu? as per docs Label= is *not* considered for matching, but exclusively used to initialize the label when nothing is set?

**Comment by DaanDeMeyer** [comment]
Either that's not working as expected or something else is going wrong, but I'm definitely getting repart failures after updating with sysupdate

**Comment by DaanDeMeyer** [comment]
repart has zero debug logging on the size algorithm it seems so I can't provide any extra useful information:

```
Can't fit requested partitions into available free space (0B), refusing.
Automatically determined minimal disk image size as 465.7G, current image size is 465.7G.
```

**Comment by DaanDeMeyer** [comment]
I actually fixed this one already with #34540
